src/operations/data_objects/index.ts

```diff
@@ -6,195 +6,239 @@ import assert from 'assert'
-            return { status: error.response?.status!, data: null }
+            message = `Error: ${error.response?.statusText}`
+            retData = { status: error.response?.status!, data: null }
         }
+        if (this.debug) console.log(message)
+        return retData
     }
 
     async replicate(
         params: DataObjectTypes.DataObjectReplicateParams
     ): Promise<
         HTTPResponse<null | DataObjectTypes.DataObjectReplicateResponse>
     > {
+        let retData
+        let message
         try {
             const res = await this.client.post(
                 '/data-objects',
                 toURLSearchParams({ op: 'replicate', ...params })
             )
-            return { status: res.status, data: res.data }
+            retData = { status: res.status, data: res.data }
         } catch (error) {
             assert(error instanceof AxiosError)
-            console.error('Error: ', error.response?.statusText)
-            return { status: error.response?.status!, data: null }
+            message = `Error: ${error.response?.statusText}`
+            retData = { status: error.response?.status!, data: null }
         }
+        if (this.debug) console.log(message)
+        return retData
     }
 
     async trim(
         params: DataObjectTypes.DataObjectTrimParams
     ): Promise<HTTPResponse<null | DataObjectTypes.DataObjectTrimResponse>> {
+        let retData
+        let message
         try {
             const res = await this.client.post(
                 '/data-objects',
                 toURLSearchParams({ op: 'trim', ...params })
             )
-            return { status: res.status, data: res.data }
+            retData = { status: res.status, data: res.data }
         } catch (error) {
             assert(error instanceof AxiosError)
-            console.error('Error: ', error.response?.statusText)
-            return { status: error.response?.status!, data: null }
+            message = `Error: ${error.response?.statusText}`
+            retData = { status: error.response?.status!, data: null }
         }
+        if (this.debug) console.log(message)
+        return retData
     }
 
     async register(
         params: DataObjectTypes.DataObjectRegisterParams
     ): Promise<
         HTTPResponse<null | DataObjectTypes.DataObjectRegisterResponse>
     > {
+        let retData
+        let message
         try {
             const res = await this.client.post(
                 '/data-objects',
                 toURLSearchParams({ op: 'register', ...params })
             )
-            return { status: res.status, data: res.data }
+            retData = { status: res.status, data: res.data }
         } catch (error) {
             assert(error instanceof AxiosError)
-            console.error('Error: ', error.response?.statusText)
-            return { status: error.response?.status!, data: null }
+            message = `Error: ${error.response?.statusText}`
```

Comment:
Yes - the string would resolve to "Error: Undefined"

Reply:
Hmm, that seems like it could lead to confusion.

Someone may think there's a bug due to `Undefined` appearing in the output.

You agree?

Reply:
Yes, I agree that would lead to confusion. However, I believe this should no longer be an issue as I changed this code in a later commit, and the message should be guaranteed to not have an undefined value in it, as the `message` field of `error` doesn't have an optional type.

https://github.com/farbs03/irods_client_http_typescript/blob/c5e936e20877e0d248eb589da42d8ecf86b9fd04/src/operations/collections/index.ts#L36

Reply:
I like that answer.

Please mark this conversation as resolved.